rfc/egress-with-ucan.md

```diff
@@ -0,0 +1,159 @@
+# Egress, authorized by UCAN
+
+## Goals
+
+* We (Storacha) want to charge users for data egress when retrieving data through our Gateway.
+* We want to provide the same free option we offer today, with a rate-limit.
+* We want to continue to provide access through Bitswap, but make it optional.
+* We eventually want to allow Storage Nodes to offer their own direct egress, bypassing our Gateway, and to charge the customer according to an agreement with them.
+
+## Abstract
+
+A proposed means for offering content egress from the Storacha Network which can be selectively authorized by UCAN, and which can track egress to bill appropriate customers. As usual within the Storacha Network, we begin from self-sovereignty: a Space definitionally has the authority to do anything with its content.
+
+To make content available as on a traditional CDN, an Agent acts on authority of a Space to make that Space's contents, or some specific CID in it, available using a bearer token, an opaque, unguessable string. The Gateway then responds to requests which contain a token by validating the proof chain, finding the content, charging for egress, and proxying it to the requester.
+
+This process does *not* include making and tracking Location Commitments. A Location Commitment is an attestation by a Storage Node that it holds a particular piece of content on behalf of a particular Space, and that it can provide it. In this proposal, we assume such a system already exists.
```

Comment:
One interesting thing to consider is that if the location commitment were to be cited in a `space/content/retrieve` delegation, in the happy path (where the content has not been moved), the gateway would already know where to fetch the content from, simply by reading the delegation that authorizes retrieval.

Probably not worthwhile but just thought I'd write it down :)